### PR TITLE
Create Logrotate Definitions to Ensure Disk Space

### PR DESCRIPTION
**Pull request by cthulhuplus**
PR for creating logrotate definitions for the bfd-server logs located here, Nathan Dister determined that the extra disk space was not needed if we rotate the logs:

/usr/local/bfd-server/access.json
/usr/local/bfd-server/access.log
/usr/local/bfd-server/bfd-server-startup.log
/usr/local/bfd-server/bluebutton-server-app-log.json
/usr/local/bfd-server/gc.log
/usr/local/bfd-server/newrelic_agent.log

### CODE REVIEW

**Comment by cthulhuplus** [comment]
Testing log rotate:
Before:
```
sudo ls -al /usr/local/bfd-server/
total 599136
drwxr-xr-x.  4 bfd  bfd       4096 Mar 16 13:03 .
drwxr-xr-x. 14 root root       160 Mar 16 12:55 ..
-rw-r--r--.  1 bfd  bfd  367001600 Mar 16 14:18 access.json
-rw-r--r--.  1 bfd  bfd        525 Mar 16 13:03 access.log
-rw-r--r--.  1 bfd  bfd    5050704 Mar 16 12:55 bfd-server-launcher-1.0.0-SNAPSHOT-capsule-fat.jar
-rw-r--r--.  1 bfd  bfd        224 Mar 16 13:03 bfd-server-startup.log
-rw-r--r--.  1 bfd  bfd   71484679 Mar 16 12:55 bfd-server-war-1.0.0-SNAPSHOT.war
-rwxrwxr-x.  1 bfd  bfd       4355 Mar 16 13:02 bfd-server.sh
-rw-------.  1 bfd  bfd       3861 Mar 16 13:02 bluebutton-appserver-keystore.jks
drwxrwxr-x.  2 bfd  bfd        209 Mar 16 13:02 bluebutton-appserver-truststore-work
-rw-rw-r--.  1 bfd  bfd       5844 Mar 16 13:02 bluebutton-appserver-truststore.jks
-rw-------.  1 bfd  bfd       5549 Mar 16 13:02 bluebutton-backend-test-data-server-client-test-keypair.pem
-rw-r--r--.  1 bfd  bfd     107454 Mar 16 14:03 bluebutton-server-app-log.json
-rw-r--r--.  1 bfd  bfd       5162 Mar 16 13:21 gc.log
drwxr-xr-x.  3 bfd  bfd         44 Mar 16 12:55 newrelic
-rw-r--r--.  1 bfd  bfd       7815 Mar 16 13:03 newrelic_agent.log
```

After:
```
sudo ls -la /usr/local/bfd-server/
total 74780
drwxr-xr-x.  4 bfd  bfd      4096 Mar 16 14:20 .
drwxr-xr-x. 14 root root      160 Mar 16 12:55 ..
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 access.json
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 access.log
-rw-r--r--.  1 bfd  bfd   5050704 Mar 16 12:55 bfd-server-launcher-1.0.0-SNAPSHOT-capsule-fat.jar
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 bfd-server-startup.log
-rw-r--r--.  1 bfd  bfd  71484679 Mar 16 12:55 bfd-server-war-1.0.0-SNAPSHOT.war
-rwxrwxr-x.  1 bfd  bfd      4355 Mar 16 13:02 bfd-server.sh
-rw-------.  1 bfd  bfd      3861 Mar 16 13:02 bluebutton-appserver-keystore.jks
drwxrwxr-x.  2 bfd  bfd       209 Mar 16 13:02 bluebutton-appserver-truststore-work
-rw-rw-r--.  1 bfd  bfd      5844 Mar 16 13:02 bluebutton-appserver-truststore.jks
-rw-------.  1 bfd  bfd      5549 Mar 16 13:02 bluebutton-backend-test-data-server-client-test-keypair.pem
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 bluebutton-server-app-log.json
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 gc.log
drwxr-xr-x.  3 bfd  bfd        44 Mar 16 12:55 newrelic
-rw-r--r--.  1 bfd  bfd         0 Mar 16 14:20 newrelic_agent.log
```

File: ops/terraform/modules/resources/templates/fhir_server.tpl [hunk 1]
```diff
@@ -4,10 +4,12 @@ set -e
 exec > >(tee -a /var/log/user_data.log 2>&1)
 
 # Extend gold image defined root partition with all available free space
-sudo growpart /dev/nvme0n1 2
-sudo pvresize /dev/nvme0n1p2
-sudo lvextend -l +100%FREE /dev/VolGroup00/rootVol
-sudo xfs_growfs /
+# The extend gold image has been commented out in favor of rotating the logs
```
Comment (jzulim) [inline ops/terraform/modules/resources/templates/fhir_server.tpl]: Maybe link or reference JIRA ticket here?

Comment (cthulhuplus) [reply ops/terraform/modules/resources/templates/fhir_server.tpl]: Done

**Comment by jzulim** [comment]
Looks like there's more to explore on the best approach with log rotation as it relates to cloud watch and Splunk, so that we don't miss any data. I'm interested to see where this goes.

File: ops/terraform/modules/resources/templates/fhir_server.tpl [hunk 2]
```diff
@@ -4,6 +4,8 @@ set -e
 exec > >(tee -a /var/log/user_data.log 2>&1)
 
 # Extend gold image defined root partition with all available free space
+# The extend gold image has been commented out in favor of rotating the logs
+# Doing this should improve deploy time, BLUEBUTTON-1582
 sudo growpart /dev/nvme0n1 2
 sudo pvresize /dev/nvme0n1p2
 sudo lvextend -l +100%FREE /dev/VolGroup00/rootVol
```
[the remark this answers is not in the document]
Comment (cthulhuplus) [reply ops/terraform/modules/resources/templates/fhir_server.tpl]: I have removed the sudo lines and successfully built. I had some tests before where it was failing to build, but it looks like it was failing for some other reason.

**Pull request merged**